### PR TITLE
[NFC] Cpp mangle use early returns and reduce indentation

### PR DESCRIPTION
**Pull request by thewilsonator**
Best reviewed with whitespace off

### CODE REVIEW

**Comment by thewilsonator** [comment]
Gah crap, I've missed something here.

**Comment by thewilsonator** [comment]
That should be windows fixed.

File: src/dmd/cppmangle.d [hunk 1]
```diff
@@ -864,74 +866,75 @@ private final class CppMangleVisitor : Visitor
...
-                    }
-                    else if (s.ident == Id.basic_iostream)
-                    {
-                        if (char_std_char_traits_char(ti, "Sd"))
-                            return;
-                    }
-                    buf.writestring("St");
-                    source_name(se, true);
-                }
+                prefix_name(p);
```
Comment (thewilsonator) [inline src/dmd/cppmangle.d:884]: will fix indent soon

**Comment by thewilsonator** [comment]
Should be green now.

**Comment by thewilsonator** [comment]
There we go, finally.

File: src/dmd/cppmangle.d [hunk 2]
```diff
@@ -600,51 +599,54 @@ private final class CppMangleVisitor : Visitor
...
+        if (!ti)
         {
-            bool needsTa = false;
+            this.writeNamespace(s.cppnamespace, () {
+                this.writeIdentifier(s.ident);
+                this.abiTags.writeSymbol(s, this);
+                },
+                haveNE);
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (thewilsonator) [reply src/dmd/cppmangle.d:613]: Generally I agree, but for one or two statement early returns I think this is better.

File: src/dmd/cppmanglewin.d [hunk 3]
```diff
@@ -909,74 +909,72 @@ extern(D):
...
         }
-        if (TemplateInstance ti = sym.isTemplateInstance())
+        auto ti = sym.isTemplateInstance();
+        if (!ti)
         {
-            auto id = ti.tempdecl.ident;
-            auto symName = id.toString();
+            writeName(sym.ident);
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (thewilsonator) [reply src/dmd/cppmanglewin.d:917]: Ditto

**Review by kinke — APPROVED** [review]

**Pull request merged**